nexus_constructor/component_type.py

```diff
@@ -37,7 +37,7 @@
     "NXslit",
     "NXvelocity_selector",
     "NXxraylens",
-]
```

Comment:
Explanation for this change: A test I wrote earlier was failing because `NXdetector_module` was placed in this list twice. This caused a mismatch between the length of the list and the number of elements in the drop-down list in the Add Component Dialog. Changed the list to a set and removed the extra `NXdetector_module` to be safe.